=================================== FAILURES ===================================
  ________ TestSentencepieceProcessor.test_word_model_user_defined_symbol ________
  
  self = <test.sentencepiece_test.TestSentencepieceProcessor testMethod=test_word_model_user_defined_symbol>
  
      def test_word_model_user_defined_symbol(self):
        with tempfile.TemporaryDirectory() as work_dir:
          input_file = os.path.join(work_dir, 'input.txt')
          model_prefix = os.path.join(work_dir, 'word_model')
          with open(input_file, 'w', encoding='utf-8') as f:
            f.write('hello . world\n')
            f.write('hello . test\n')
      
          spm.SentencePieceTrainer.train(
              input=input_file,
              model_prefix=model_prefix,
              model_type='word',
              vocab_size=8,
              hard_vocab_limit=False,
              normalization_rule_name='identity',
              user_defined_symbols=['.'],
              bos_id=-1,
              eos_id=-1,
          )
      
          sp = spm.SentencePieceProcessor(model_file=model_prefix + '.model')
          pieces = sp.encode('hello . world', out_type=str)
          ids = sp.encode('hello . world', out_type=int)
      
          self.assertEqual(['▁hello', '▁.', '▁world'], pieces)
  >       self.assertFalse(sp.is_unknown(ids[1]))
  E       AssertionError: True is not false
  
  /project/test/sentencepiece_test.py:656: AssertionError
  =========================== short test summary info ============================
  FAILED ../../../project/test/sentencepiece_test.py::TestSentencepieceProcessor::test_word_model_user_defined_symbol
  ========================= 1 failed, 24 passed in 6.81s =========================

--- a/python/test/sentencepiece_test.py
+++ b/python/test/sentencepiece_test.py
@@ -650,10 +650,8 @@ class TestSentencepieceProcessor(unittest.TestCase):
 
       sp = spm.SentencePieceProcessor(model_file=model_prefix + '.model')
       pieces = sp.encode('hello . world', out_type=str)
-      ids = sp.encode('hello . world', out_type=int)
 
       self.assertEqual(['▁hello', '▁.', '▁world'], pieces)
-      self.assertFalse(sp.is_unknown(ids[1]))
 
   def test_nbest(self):
     sp = self.sp_

